

1. Scenario: A company wants to analyze the sales performance of its products in different regions. They have collected the following data:
   Region A: [10, 15, 12, 8, 14]
   Region B: [18, 20, 16, 22, 25]
   Calculate the mean sales for each region.

In [1]:
def calculate_mean_sales(region_sales):
    total_sales = sum(region_sales)
    num_sales = len(region_sales)
    
    mean_sales = total_sales / num_sales
    return mean_sales

# Sales data for each region
region_a_sales = [10, 15, 12, 8, 14]
region_b_sales = [18, 20, 16, 22, 25]

# Calculate the mean sales for each region
mean_sales_region_a = calculate_mean_sales(region_a_sales)
mean_sales_region_b = calculate_mean_sales(region_b_sales)

# Print the results
print("Mean sales for Region A:", mean_sales_region_a)
print("Mean sales for Region B:", mean_sales_region_b)


Mean sales for Region A: 11.8
Mean sales for Region B: 20.2


2. Scenario: A survey is conducted to measure customer satisfaction on a scale of 1 to 5. The data collected is as follows:
   [4, 5, 2, 3, 5, 4, 3, 2, 4, 5]
   Calculate the mode of the survey responses.

In [1]:
from statistics import mode

data = [4, 5, 2, 3, 5, 4, 3, 2, 4, 5]
mode_value = mode(data)

print("The mode of the survey responses is:", mode_value)


The mode of the survey responses is: 4




3. Scenario: A company wants to compare the salaries of two departments. The salary data for Department A and Department B are as follows:
   Department A: [5000, 6000, 5500, 7000]
   Department B: [4500, 5500, 5800, 6000, 5200]
   Calculate the median salary for each department.

In [3]:
from statistics import median

Department_A = [5000, 6000, 5500, 7000]
Department_B = [4500, 5500, 5800, 6000, 5200]

data1 = median(Department_A)
data2 = median(Department_B)

print("THe mode of Department A:", data1, "\nThe mode of Department B:", data2)

THe mode of Department A: 5750.0 
The mode of Department B: 5500




4. Scenario: A data analyst wants to determine the variability in the daily stock prices of a company. The data collected is as follows:
   [25.5, 24.8, 26.1, 25.3, 24.9]
   Calculate the range of the stock prices.

In [4]:
data = [25.5, 24.8, 26.1, 25.3, 24.9]

range = max(data) - min(data)

print("The range of data:", range)

The range of data: 1.3000000000000007




5. Scenario: A study is conducted to compare the performance of two different teaching methods. The test scores of the students in each group are as follows:
   Group A: [85, 90, 92, 88, 91]
   Group B: [82, 88, 90, 86, 87]
   Perform a t-test to determine if there is a significant difference in the mean scores between the two groups.

In [6]:
from scipy import stats

# Test scores for Group A and Group B
group_a = [85, 90, 92, 88, 91]
group_b = [82, 88, 90, 86, 87]

# Perform t-test
t_statistic, p_value = stats.ttest_ind(group_a, group_b)

# Check if there is a significant difference
alpha = 0.05  # Significance level
if p_value < alpha:
    result = "There is a significant difference between the mean scores of the two groups."
else:
    result = "There is no significant difference between the mean scores of the two groups."

# Print the result
print(result)


There is no significant difference between the mean scores of the two groups.



6. Scenario: A company wants to analyze the relationship between advertising expenditure and sales. The data collected is as follows:
   Advertising Expenditure (in thousands): [10, 15, 12, 8, 14]
   Sales (in thousands): [25, 30, 28, 20, 26]
   Calculate the correlation coefficient between advertising expenditure and sales.

In [7]:
import numpy as np

ad_expenditure = [10, 15, 12, 8, 14]
sales = [25, 30, 28, 20, 26]

# Calculate correlation coefficient
correlation_coefficient = np.corrcoef(ad_expenditure, sales)[0, 1]

# Print the correlation coefficient
print("The correlation coefficient between advertising expenditure and sales is:", correlation_coefficient)

The correlation coefficient between advertising expenditure and sales is: 0.8757511375750132


7. Scenario: A survey is conducted to measure the heights of a group of people. The data collected is as follows:
   [160, 170, 165, 155, 175, 180, 170]
   Calculate the standard deviation of the heights.

In [8]:
#Solution:

import numpy as np

heights = [160, 170, 165, 155, 175, 180, 170]

std_dev = np.std(heights)

# Print the standard deviation
print("The standard deviation of the heights is:", std_dev)

The standard deviation of the heights is: 7.953949089757174


8. Scenario: A company wants to analyze the relationship between employee tenure and job satisfaction. The data collected is as follows:
   Employee Tenure (in years): [2, 3, 5, 4, 6, 2, 4]
   Job Satisfaction (on a scale of 1 to 10): [7, 8, 6, 9, 5, 7, 6]
   Perform a linear regression analysis to predict job satisfaction based on employee tenure.

In [9]:
from scipy import stats

employee_tenure = [2, 3, 5, 4, 6, 2, 4]
job_satisfaction = [7, 8, 6, 9, 5, 7, 6]

# Perform linear regression analysis
slope, intercept, r_value, p_value, std_err = stats.linregress(employee_tenure, job_satisfaction)

# Print the regression results
print("Slope (Regression Coefficient):", slope)
print("Intercept:", intercept)
print("R-squared:", r_value**2)
print("p-value:", p_value)
print("Standard Error:", std_err)


Slope (Regression Coefficient): -0.4680851063829787
Intercept: 8.595744680851062
R-squared: 0.27099664053751393
p-value: 0.23095550952297988
Standard Error: 0.34333860006713723




9. Scenario: A study is conducted to compare the effectiveness of two different medications. The recovery times of the patients in each group are as follows:
   Medication A: [10, 12, 14, 11, 13]
   Medication B: [15, 17, 16, 14, 18]
   Perform an analysis of variance (ANOVA) to determine if there is a significant difference in the mean recovery times between the two medications.

In [12]:
from scipy import stats

# Recovery times for Medication A and Medication B
medication_a_times = [10, 12, 14, 11, 13]
medication_b_times = [15, 17, 16, 14, 18]

# Perform ANOVA
f_value, p_value = stats.f_oneway(medication_a_times, medication_b_times)

print("f_value",f_value, "\np_value", p_value)
# Check if there is a significant difference
alpha = 0.05  # Significance level
if p_value < alpha:
    result = "There is a significant difference in the mean recovery times between the two medications."
else:
    result = "There is no significant difference in the mean recovery times between the two medications."

# Print the result
print(result)


f_value 16.0 
p_value 0.003949772803445326
There is a significant difference in the mean recovery times between the two medications.




10. Scenario: A company wants to analyze customer feedback ratings on a scale of 1 to 10. The data collected is

 as follows:
    [8, 9, 7, 6, 8, 10, 9, 8, 7, 8]
    Calculate the 75th percentile of the feedback ratings.


In [14]:
import numpy as np

feedback_ratings = [8, 9, 7, 6, 8, 10, 9, 8, 7, 8]

# Calculate the 75th percentile
percen75 = np.percentile(feedback_ratings, 75)

# Print the 75th percentile
print("The 75th percentile of the feedback ratings is:", percen75)

The 75th percentile of the feedback ratings is: 8.75



11. Scenario: A quality control department wants to test the weight consistency of a product. The weights of a sample of products are as follows:
    [10.2, 9.8, 10.0, 10.5, 10.3, 10.1]
    Perform a hypothesis test to determine if the mean weight differs significantly from 10 grams.



12. Scenario: A company wants to analyze the click-through rates of two different website designs. The number of clicks for each design is as follows:
    Design A: [100, 120, 110, 90, 95]
    Design B: [80, 85, 90, 95, 100]
    Perform a chi-square test to determine if there is a significant difference in the click-through rates between the two designs.



13. Scenario: A survey is conducted to measure customer satisfaction with a product on a scale of 1 to 10. The data collected is as follows:
    [7, 9, 6, 8, 10, 7, 8, 9, 7, 8]
    Calculate the 95% confidence interval for the population mean satisfaction score.



14. Scenario: A company wants to analyze the effect of temperature on product performance. The data collected is as follows:
    Temperature (in degrees Celsius): [20, 22, 23, 19, 21]
    Performance (on a scale of 1 to 10): [8, 7, 9, 6, 8]
    Perform a simple linear regression to predict performance based on temperature.



15. Scenario: A study is conducted to compare the preferences of two groups of participants. The preferences are measured on a Likert scale from 1 to 5. The data collected is as follows:
    Group A: [4, 3, 5, 2, 4]
    Group B: [3, 2, 4, 3, 3]
    Perform a Mann-Whitney U test to determine if there is a significant difference in the median preferences between the two groups.



16. Scenario: A company wants to analyze the distribution of customer ages. The data collected is as follows:
    [25, 30, 35, 40, 45, 50, 55, 60, 65, 70]
    Calculate the interquartile range (IQR) of the ages.



17. Scenario: A study is conducted to compare the performance of three different machine learning algorithms. The accuracy scores for each algorithm are as follows:
    Algorithm A: [0.85, 0.80, 0.82, 0.87, 0.83]
    Algorithm B: [0.78, 0.82, 0.84, 0.80, 0.79]
    Algorithm C: [0.90, 0.88, 0.89, 0.86, 0.87]
    Perform a Kruskal-Wallis test to determine if there is a significant difference in the median accuracy scores between the algorithms.



18. Scenario: A company wants to analyze the effect of price on sales. The data collected is as follows:
    Price (in dollars): [10, 15, 12, 8, 14]
    Sales: [100, 80, 90, 110, 95]
    Perform a simple linear regression to predict

 sales based on price.



19. Scenario: A survey is conducted to measure the satisfaction levels of customers with a new product. The data collected is as follows:
    [7, 8, 9, 6, 8, 7, 9, 7, 8, 7]
    Calculate the standard error of the mean satisfaction score.



20. Scenario: A company wants to analyze the relationship between advertising expenditure and sales. The data collected is as follows:
    Advertising Expenditure (in thousands): [10, 15, 12, 8, 14]
    Sales (in thousands): [25, 30, 28, 20, 26]
    Perform a multiple regression analysis to predict sales based on advertising expenditure.
